contracts/ColonyFunding.sol

```diff
@@ -141,6 +142,86 @@ contract ColonyFunding is ColonyStorage, DSMath {
     return nonRewardPotsTotal[_token];
   }
 
+  function getGlobalRewardPayoutCount() public view returns (uint256) {
+    return globalRewardPayoutCount;
+  }
+
+  function getUserRewardPayoutCount(address _user) public view returns (uint256) {
+    return userRewardPayoutCount[_user];
+  }
+
+  function startNextRewardPayout(address _token) public auth returns(uint256) {
+    globalRewardPayoutCount += 1;
+
+    //TODO: Lock everyones tokens
+
+    uint256 amount = pots[0].balance[_token];
+    pots[0].balance[_token] = 0;
+
+    rewardPayoutCycles[globalRewardPayoutCount] = RewardPayoutCycle(
+      IColonyNetwork(colonyNetworkAddress).getReputationRootHash(),
+      amount,
+      amount,
+      _token,
+      block.number
+    );
+
+    return globalRewardPayoutCount;
+  }
+
+  function claimRewardPayout(uint8 _payoutId, int256 _userReputation, int256 _totalReputation) public {
+    require(block.number - rewardPayoutCycles[_payoutId].blockNumber < 500000);
+    require(_payoutId - userRewardPayoutCount[msg.sender] == 1);
+
+    //TODO: Prove that userReputation and totalReputation in reputationState are correct
+
+    address tokenAddress = rewardPayoutCycles[_payoutId].token;
+    ERC20Extended payoutToken = ERC20Extended(tokenAddress);
+    uint256 userTokens = payoutToken.balanceOf(msg.sender);
+    uint256 totalTokens = sub(payoutToken.totalSupply(), payoutToken.balanceOf(address(this)));
+
+    require(userTokens > 0 && totalTokens > 0 && _userReputation > 0 && _totalReputation > 0);
+
+    uint256 numOfDigits = SafeMath.numDigits(totalTokens * uint256(_totalReputation));
+    uint256 factor = 10 ** (numOfDigits / 2);
+
+    uint256 sharePercentage = SafeMath.sqrt(userTokens * uint256(_userReputation) * factor / (totalTokens * uint256(_totalReputation) / factor));
+    uint256 reward = rewardPayoutCycles[_payoutId].amount * sharePercentage / factor;
+
+    require(rewardPayoutCycles[_payoutId].remaining >= reward);
+
+    userRewardPayoutCount[msg.sender] += 1;
+
+    //TODO: Unlock user tokens
+
+    rewardPayoutCycles[_payoutId].remaining = sub(rewardPayoutCycles[_payoutId].remaining, reward);
+
+    payoutToken.transfer(msg.sender, reward);
+  }
+
+  function waiveRewardPayout(uint8 _payoutId) public {
+    require(_payoutId - userRewardPayoutCount[msg.sender] == 1);
```

Comment:
Arguably, it should be possible for a user to waive multiple payouts at once - if they've been away for a long time, maybe there are lots for them to waive.